visor/init.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2022 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package visor
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"code.vegaprotocol.io/vega/logging"
+	"code.vegaprotocol.io/vega/visor/config"
+	"code.vegaprotocol.io/vega/visor/utils"
+)
+
+func Init(log *logging.Logger, homeFolder string) error {
+	homePath, err := utils.AbsPath(homeFolder)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path for %q: %w", homeFolder, err)
+	}
+
+	homeExists, err := utils.PathExists(homePath)
+	if err != nil {
+		return fmt.Errorf("failed to check if %q path exists: %w", homePath, err)
+	}
+
+	if homeExists {
+		return fmt.Errorf("home folder %q already exists", homePath)
+	}
+
+	visorConf := config.DefaultVisorConfig(log, homePath)
+
+	log.Info("Initiating genesis folder")
+	if err := initDefaultFolder(visorConf.GenesisFolder(), "genesis"); err != nil {
```

Comment:
what is this supposed to be doing? it it used for anything?

Reply:
This will generate basic folder structure for the tool `home` with example configs.

Reply:
but it's not used functionally as far as I could see, maybe I've missed it

Reply:
Ah ok.. I see what you mean.. Yeah it is not used here in the code but it is used by the `init` command.

Capsule for example uses the init command [here](https://github.com/vegaprotocol/vegacapsule/pull/210/files#diff-d44b310e81776ebbb501940b14f82b9dec20d1d458f9b5710c36423fae69b3bbR40)